capstone/src/test/java/com/google/step/search/ElasticsearchClientTest.java

```diff
@@ -176,8 +184,21 @@ public LowLevelHttpResponse execute() throws IOException {
       }
     };
 
+    MockHttpTransport transportSpy = spy(transport);
     ElasticsearchClient esClient =
-        new ElasticsearchClient(transport, elasticsearchHostname, elasticsearchPort);
-    assertEquals(Arrays.asList(HEADER_1, HEADER_2), esClient.getRandomRestaurants());
+        new ElasticsearchClient(transportSpy, elasticsearchHostname, elasticsearchPort);
+
+    String expectedUrl =
+        new URIBuilder()
+            .setScheme("http")
+            .setHost(elasticsearchHostname)
+            .setPort(elasticsearchPort)
+            .setPath("/_search")
+            .toString();
+
+    List<RestaurantHeader> searchResults = esClient.searchRestaurants("");
+
+    verify(transportSpy).buildRequest("POST", expectedUrl);
```

Comment:
I think this is a good approach here. Nice work!

Reply:
+1 Nice job!